test/Text/Markdown/UnlitSpec.hs

```diff
@@ -33,20 +34,22 @@ spec = do
 
     it "unlits code marked with .haskell by default (unless it is marked with .ignore as well)" $ do
       withTempFile $ \infile -> withTempFile $ \outfile -> do
-        writeFile infile . build $ do
-          "~~~ {.haskell}"
-          "some code"
-
-          "~~~"
-          "~~~ {.haskell .ignore}"
-          "some other code"
-
-          "~~~"
-        run ["-h", "Foo.lhs", infile, outfile]
-        readFile outfile `shouldReturn` (build $ do
-          "#line 2 \"Foo.lhs\""
-          "some code"
-          )
+        let withEsc esc = do
+              writeFile infile $ unlines
+                  [ esc ++ " {.haskell}"
+                  , "some code"
+                  , esc
+                  , esc ++ " {.haskell .ignore}"
+                  , "some other code"
+                  , esc
+                  ]
+              run ["-h", "Foo.lhs", infile, outfile]
+              readFile outfile `shouldReturn` (build $ do
+                "#line 2 \"Foo.lhs\""
+                "some code"
+                )
+        withEsc "~~~"
+        withEsc "```"
```

Comment:
I think the right level to test this is on `parse` rather than as an acceptance test on `run`.  I did that in https://github.com/sol/markdown-unlit/commit/01e16623b3218c60c865e7cb32c79a1bdec81f3a and pushed your other commit to `master`.